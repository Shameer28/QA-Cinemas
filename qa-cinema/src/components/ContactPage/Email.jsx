import React, { useState } from "react";
import "bootstrap/dist/css/bootstrap.min.css";
import Card from 'react-bootstrap/Card'
import Container from "react-bootstrap/Container";
import Form from "react-bootstrap/Form";
import { Button } from "react-bootstrap";

const Email = () => {
    const [Email, setEmail] = useState('');
    const [FirstName, setFirstName] = useState('');
    const [LastName, setLastName] = useState('');
    const [TextArea, setTextArea] = useState('');




    const handleSubmit = (e) => {
        //Preventing the page from refreshing
        e.preventDefault();

        const dataToSend = JSON.stringify({ Email, FirstName, LastName, TextArea });
        console.log("data has been sent");
            

        //Setting the states back to ''
        setEmail('');
        setFirstName('');
        setLastName('');
        setTextArea('');


        // const handleSubmits = (e) => {
        //     e.preventDefault();
        //     axios({
        //         method: "POST",
        //         url: "http://localhost:3000/send",
        //         data: this.data
        //     }).then((response) => {
        //         if (response.data.status === 'success') {
        //             alert("Message Sent.");
        //             this.resetForm()
        //         } else if (response.data.status === 'fail') {
        //             alert("Message failed to send.")
        //         }
        //     });
        // };

    }
    return (
        <div style={{ width: "100%", backgroundColor: "#212529", color: "white" }}>
            <Container>
                <Card className="text-center" style={{ backgroundColor: "#212529", color: "white" }}>
                    <Card.Header>Contact US</Card.Header>
                    <Card.Body>
                        <Card.Title><u>Please fill out the form to submit a request</u></Card.Title>
                        <Card.Text>
                            <Form onSubmit={handleSubmit} >
                                <Form.Group>
                                    <Form.Label>Email address</Form.Label>
                                    <Form.Control id="Email" type="email" className="form-control" placeholder="name@example.com" value={Email} required onChange={(e) => { setEmail(e.target.value) }} />
                                </Form.Group>
                                <Form.Group >
                                    <Form.Label>First Name</Form.Label>
                                    <Form.Control id="FirstName" type="text" className="form-control" required minLength={3} placeholder="First Name" value={FirstName} onChange={(e) => { setFirstName(e.target.value) }} />

                                    <Form.Label>Last Name</Form.Label>
                                    <Form.Control id="LastName" type="text" className="form-control" required minLength={3} placeholder="Last Name" value={LastName} onChange={(e) => { setLastName(e.target.value) }} />
                                </Form.Group>
                                <Form.Group >
                                    <Form.Label>Message</Form.Label>
                                    <Form.Control id="TextArea" as="textarea" className="form-control" required minLength={3} rows={3} value={TextArea} onChange={(e) => { setTextArea(e.target.value) }} />
                                </Form.Group>
                                <br />
                                <Button type="submit"> Submit</Button>
                            </Form>
                        </Card.Text>

                    </Card.Body>
                    <Card.Footer className="text-muted"></Card.Footer>
                </Card>
            </Container>
        </div>
    );
}
export default Email;