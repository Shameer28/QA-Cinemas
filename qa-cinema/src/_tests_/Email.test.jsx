import Email from "../components/ContactPage/Email";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe(`Form testing`,() => {

    describe(`onSubmit test`, () => {

        // Parent level arranging
        const data = "test_name";
        const email = "test@gmail.com";
        const last = "doe";
        const Text = "This is a test to see if the text area works";
        it(`Should update the value of the input (which updates state) when text is typed in`, () => {

            // ? Arrange
            // render(Form) is being deconstructed and saved as 'container'
            const {container} = render(<Email/>);

            const textInput = container.querySelector('#FirstName');
            const textInput2 = container.querySelector('#Email');
            const textInput3 = container.querySelector('#LastName');
            const textInput4 = container.querySelector('#TextArea');
            // console.log(textInput);
            expect(textInput.value).toEqual("");
            expect(textInput2.value).toEqual("");
            expect(textInput3.value).toEqual("");
            expect(textInput4.value).toEqual("");
            // ? Act - simulated typing
            userEvent.type(textInput, data);
            userEvent.type(textInput2, email);
            userEvent.type(textInput3, last);
            userEvent.type(textInput4, Text);
            // ? Assert
            expect(textInput.value).toEqual(data);
            expect(textInput2.value).toEqual(email);
            expect(textInput3.value).toEqual(last);
            expect(textInput4.value).toEqual(Text);
        });

        it(`Should call handleSubmit() when the submit button is clicked`, () => {

            const mockSubmit = jest.fn();

            // Arrange
            const {container} = render(<Email/>);

            const submitButton = container.querySelector('#submit');

            // Act
             submitButton.onClick = mockSubmit();
             userEvent.click(submitButton, mockSubmit())

            // Assert
            expect(mockSubmit).toHaveBeenCalled();

        });

    });    
})