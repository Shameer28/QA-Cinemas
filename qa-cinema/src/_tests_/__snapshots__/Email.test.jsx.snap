// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Form testing Should match the snapshot 1`] = `null`;
